src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerIndexService.kt

```diff
@@ -71,15 +71,26 @@ class PrisonerIndexService(
   fun syncPrisoner(prisonerId: String): Prisoner? =
     nomisService.getOffender(prisonerId)?.let {
       reindex(offenderBooking = it)
-    } ?: run {
-      telemetryClient.trackEvent(
-        "POSOffenderNotFoundForIndexing",
-        mapOf("prisonerID" to prisonerId),
-        null,
-      )
-      log.error("POSOffenderNotFoundForIndexing {}", prisonerId)
-      return null
     }
+      ?: get(prisonerId)?.run {
+        delete(prisonerId)
+        telemetryClient.trackEvent(
+          "POSOffenderRemovedFromIndex",
```

Comment:
I added a `trackEvent` to the `delete` method, don't think we therefore need this event too.  Although I do prefer your event name, mine was named that way to match the `POSPrisonerCreated` and `POSPrisonerUpdated` events...